governor.py

```diff
@@ -0,0 +1,408 @@
+import copy
+import inspect
+import logging
+import requests
+
+from collections import defaultdict, deque
+
+from config import _is_affirmative
+
+
+log = logging.getLogger('governor')
+
+
+class LimiterConfigError(Exception):
+    """
+    Error when parsing limiter
+    """
+    pass
+
+
+class Governor(object):
+    """
+    Abstract Governor class
+    """
+
+    # Defines what's an 'active' context
+    # i.e. number of iterations for which a context (not encountered anymore)
+    # is kept stored before being flushed.
+    _CONTEXTS_TTL = 3
+
+    # Collector payload's metric format
+    _METRIC_FORMAT = ['name', 'timestamp', 'value', 'attributes']
+
+    # Governor report endpoint
+    DATADOG_REPORT_GOVERNOR_URL = 'agent_governor/report_too_many_active_contexts'
+
+    # Governor agent-specific static variables
+    _CHECK_LIMITERS, _AGENT_LIMITERS = None, None
+    _HOSTNAME = None
+    _DD_URL = None
+    _API_KEYS = None
+
+    @classmethod
+    def init(cls, governor_config, agent_config={}, hostname=None):
+        """
+        Set Governor agent-specific variables
+        """
+        if not _is_affirmative(agent_config.get('use_governor', True)):
+            return
+
+        cls._CHECK_LIMITERS, cls._AGENT_LIMITERS = LimiterParser.parse_limiters(governor_config)
+        cls._HOSTNAME = hostname
+        cls._DD_URL = agent_config.get('dd_url')
```

Comment:
You should probably use the post-formatted URL instead here? see https://github.com/DataDog/dd-agent/commit/92f0c759df0dc17d76a7d28c5cbe1e0bf159c4f8#diff-27d4010935202c09554e28f0bb3239a7R196